### DISCUSSION TITLE
Which headless ui component should I use for complex dropdown?

### DISCUSSION THREAD

**Discussion by Jun-Xiang** [tailwindlabs/headlessui#2428] [Help]
I would like to build something similar to [this](https://flowbite.com/blocks/application/filter/#advanced-filter-by-keywords) using headless ui but was unable to find a suitable component in headless ui. `Menu` makes sense to be used but based on the docs, its not suitable for this complex scenario.
> They have specific accessibility semantics, and their content should be restricted to a list of links or buttons. 

Next there is `Popover` but based on the docs written in `Menu`, it is saying that it's for navigation purposes (correct me if I am wrong).
> They're great for building header nav items with expandable content and flyout panels.

Should I use `Menu` without using the `Menu.Item` to prevent the `Menu` from auto closing or is there any alternatives that is great for this use case?

**Comment by babycourageous** [comment]
Did you wind up landing on anything?
I was actually looking at something similar. I think Popover or Dialog would be the way to go. Since that filter has an interactive header I might lean towards the Dialog.
Hope that helps!
